core/src/mast/node/join_node.rs

```diff
@@ -18,6 +19,8 @@ use crate::{
 pub struct JoinNode {
     children: [MastNodeId; 2],
     digest: RpoDigest,
+    before_enter: Vec<DecoratorId>,
```

Comment:
Looks like most places where we are using `Vec<DecoratorId>` here could use `Box<[DecoratorId]>` instead, and shave off 8 bytes per field (on 64-bit systems). Doesn't look like we ever mutate them in-place, just create/replace. AFAIK `Box::from([]) as Box<[DecoratorId]>` won't actually allocate anything.

Not something we need to change in this PR necessarily, but that'd be the first thing I'd do to reduce the size of `MastNode` when we start trying to trim down the size.